src/core/DoomInPlayCounter.ttslua

```diff
@@ -1,132 +1,105 @@
+        color = { 0, 0, 0, 0 }
+    })
+
+    -- context menu
+    self.addContextMenuItem("More Information", moreInformation)
+    self.addContextMenuItem("Remove Doom", function()
+        removeDoom = true
+        Wait.stop(loopID)
+        countDoom()
+        Wait.time(function()
+            removeDoom = false
+            loopID = Wait.time(countDoom, 2, -1)
+        end, 2)
+    end)
+
+    loopID = Wait.time(countDoom, 2, -1)
 end
 
-function swap_fcolor(_obj, _color, alt_click)
-    light_mode = not light_mode
-    reloadAll()
+function moreInformation()
+    printToAll("------------------------------", "White")
+    printToAll("Doom-in-Play Counter v" .. information["version"] .. " by Chr1Z", "Orange")
+    printToAll("last updated: " .. information["last_updated"], "White")
+    printToAll("Automatically counts the doom in play (exluding the agenda and objects with the ignore tag).", "Green")
+    printToAll("ignore tag: " .. IGNORE_TAG, "White")
 end
 
-function swap_align(_obj, _color, alt_click)
-    center_mode = not center_mode
-    reloadAll()
+-- main function
+function countDoom()
+    local doom = 0
+    for i = 1, 5 do doom = doom + search(i) end
+    self.editButton({ index = 0, label = tostring(doom) })
 end
 
-function editName(_obj, _string, value)
-    self.setName(value)
-    setTooltips()
-end
+-- searches playermats (num = 1-4) or the scripting zone (num = 5)
+function search(num)
+    local val = 0
+    if num == 5 then
+        for _, obj in ipairs(zone.getObjects()) do
+            val = val + getResult(obj)
+        end
+    else
+        CAST_PARAMETERS.origin = originAndSize[num].origin
+        CAST_PARAMETERS.size = originAndSize[num].size
 
-function add_subtract(_obj, _color, alt_click)
-    mod = alt_click and -1 or 1
-    new_value = math.min(math.max(val + mod, MIN_VALUE), MAX_VALUE)
-    if val ~= new_value then
-        val = new_value
-      updateVal()
-        updateSave()
+        for _, obj in ipairs(Physics.cast(CAST_PARAMETERS)) do
+            val = val + getResult(obj.hit_object)
+        end
     end
+    return val
 end
 
-function updateVal()
-
-    self.editButton({
-        index = 0,
-        label = tostring(val),
-
-        })
-end
-
-function reset_val()
-    val = 0
-    updateVal()
-    updateSave()
-end
-
-function setTooltips()
-    self.editInput({
-        index = 0,
-        value = self.getName(),
-        tooltip = ttText
-        })
-    self.editButton({
-        index = 0,
-        value = tostring(val),
-        tooltip = ttText
-        })
-end
-
-function null()
-end
-
-function keepSample(_obj, _string, value)
-    reloadAll()
+-- checks an object for the doom image and gets quantity (for stacks)
+function getResult(obj)
+    if (obj.is_face_down and obj.getCustomObject().image_bottom == doom_url) or
+        (obj.name == "Custom_Token" and obj.getCustomObject().image == doom_url) then
+        if not obj.hasTag(IGNORE_TAG) then
+            if removeDoom then
```

Comment:
Relying on a global variable to change behavior is going to be error prone, especially with any kind of timed/background operation.  At a minimum it should be a parameter passed here, but better to make removal a separate method and call it when appropriate.

Reply:
loop will be stopped before setting the global variable, then it will be called once and only afterwards the loop will be restarted